packages/hoppscotch-sh-admin/src/composables/useClientHandler.ts

```diff
@@ -25,22 +25,26 @@ export function useClientHandler<
 
   const fetchData = async () => {
     fetching.value = true;
-    try {
-      const result = await client
-        .query(query, {
-          ...variables,
-        })
-        .toPromise();
 
-      if (getList) {
-        const resultList = getList(result.data!);
-        dataAsList.value.push(...resultList);
-      } else {
-        data.value = result.data;
-      }
-    } catch (e) {
+    const result = await client
+      .query(query, {
+        ...variables,
+      })
+      .toPromise();
+
+    if (result.error) {
       error.value = true;
+      fetching.value = false;
+      return;
+    }
+
+    if (getList) {
+      const resultList = getList(result.data!);
+      dataAsList.value.push(...resultList);
+    } else {
+      data.value = result.data;
     }
+
     fetching.value = false;
```

Comment:
We're using the `useClientHandle` composable that never rejects (clubbing with `toPromise`) or throws but returns an object with `error` being a part, [Ref](https://commerce.nearform.com/open-source/urql/docs/api/core/#operationresult).

Reply:
@jamesgeorge007 Got it. Try catch is redundant here